utils/hv-tools/Makefile

```diff
@@ -0,0 +1,52 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=hv-tools
+PKG_VERSION:=1.0.0
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/NikulovE/hv-tools/
+PKG_SOURCE_VERSION:=e1d320d028752afe7b7c3adbfc6f91b4305d50a2
+PKG_MIRROR_HASH:=4d9a5d823df86a2c3ad59201a3ca2438a9e679e384ee59119b7599df2a30d1a1
+
+PKG_MAINTAINER:=Evgeniy Nikulov <nikulov@live.ru>
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/kernel.mk
+
+define Package/hv-tools
+  SECTION:=utils
+  CATEGORY:=Utilities
+  SUBMENU:=Virtualization
+  TITLE:=Hyper-V Guest Tools
+  DEPENDS:=+procd +libpthread @(TARGET_x86||TARGET_x86_64)
+endef
+
+define Package/hv-tools/description
+This package contains Hyper-V Guest Tools for OpenWrt, including various utilities and daemons.
+endef
+
+define Build/Compile
+	$(MAKE_VARS) $(MAKE) $(MAKE_FLAGS) all -C $(PKG_BUILD_DIR)/files/tools/hv
+endef
+
+define Package/hv-tools/install
+	$(INSTALL_DIR) $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/files/tools/hv/hv_kvp_daemon $(1)/usr/sbin/hv_kvp_daemon
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/files/tools/hv/hv_vss_daemon $(1)/usr/sbin/hv_vss_daemon
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/files/tools/hv/hv_fcopy_daemon $(1)/usr/sbin/hv_fcopy_daemon
+	
+	$(INSTALL_DIR) $(1)/usr/libexec/hypervkvpd 
+	$(INSTALL_BIN) ./files/usr/libexec/hypervkvpd/hv_get_dhcp_info.sh $(1)/usr/libexec/hypervkvpd/hv_get_dhcp_info
+	$(INSTALL_BIN) ./files/usr/libexec/hypervkvpd/hv_get_dns_info.sh $(1)/usr/libexec/hypervkvpd/hv_get_dns_info
+	$(INSTALL_BIN) ./files/usr/libexec/hypervkvpd/hv_set_ifconfig.sh $(1)/usr/libexec/hypervkvpd/hv_set_ifconfig
+	
+	$(INSTALL_DIR) $(1)/etc/init.d
+	$(INSTALL_BIN) ./files/etc/init.d/hv_fcopy_daemon $(1)/etc/init.d/hv_fcopy_daemon
+	$(INSTALL_BIN) ./files/etc/init.d/hv_kvp_daemon $(1)/etc/init.d/hv_kvp_daemon
+	$(INSTALL_BIN) ./files/etc/init.d/hv_vss_daemon $(1)/etc/init.d/hv_vss_daemon
```

Comment:
I am not sure what you want to achieve here. These files you want to add here and also, these files are in your repository. 🤷  Why we should have these files twice?

Reply:
Script files in /usr/libexec/hypervkvpd/:
These shell scripts (hv_get_dhcp_info.sh, hv_get_dns_info.sh, and hv_set_ifconfig.sh) are essential for the functioning of the hv_kvp_daemon service. They are responsible for gathering and setting network configuration data, which is required for proper integration with Hyper-V. Placing them in /usr/libexec/hypervkvpd/ makes them accessible to the hv_kvp_daemon during runtime.

Init scripts in /etc/init.d/:
These init scripts (hv_fcopy_daemon, hv_kvp_daemon, and hv_vss_daemon) are necessary for managing the lifecycle (start, stop, restart) of the corresponding daemons. They ensure that these services are correctly initialized when the system starts. Including them in /etc/init.d/ ensures that they are recognized and handled properly by OpenWrt's service management system.
The files in the repository are the source files, whereas the Makefile instructions are about installing these files in the correct locations on the target system during package installation. This approach follows common package management practices where source files are kept in the repository, and the Makefile defines how they are deployed on the system.

+ It`s already in Linux Kernel https://github.com/torvalds/linux/tree/master/tools/hv I just pack it into package and bring back functionality

Reply:
Ok, understand.  I will switch to copy files only from remote repo

Reply:
In your deleted comment, there was one great point. These files are already present in the Linux kernel repository. Why have you fork it? Look at other Makefiles, how it is done and use it. No need to reinvent a wheel once again.

Files are located here: https://github.com/torvalds/linux/tree/master/tools/hv

Reply:
I deleted because at first glance I understand your question like "What the difference between two similar files?" but after deep reading understand that you talk about to copy files from remote repo instead duplicate files in remote repo and this "packages" repo. 
Yes, this files already in Linux kernel.  I also found these files in OpenWrt folder structure during building, but one file using network manager (https://github.com/torvalds/linux/blob/master/tools/hv/hv_set_ifconfig.sh) -so I rework it. 
I tried to use Kernel`s source and It was too hard to me in steps hv-tools/check, hv-tools/compile - it`s require to bring all kernel`s source before "compile" step. In case to compile Openwrt firmware for specific device is not a problem, but for PR`s checks it should independed as I understand. 
@yichya also tried to make this package https://github.com/yichya/openwrt-hv-tools/blob/master/Makefile I don`t know why he didn`t finished PR as contributor of OpenWrt.